CalibTracker/SiPixelESProducers/plugins/SiPixelFakeGenErrorDBObjectESSource.cc

```diff
@@ -35,9 +34,9 @@ std::unique_ptr<SiPixelGenErrorDBObject> SiPixelFakeGenErrorDBObjectESSource::pr
   //  open the GenError file(s)
   for (m = 0; m < obj->numOfTempl(); ++m) {
     edm::FileInPath file(GenErrorCalibrations_[m].c_str());
-    tempfile = (file.fullPath()).c_str();
+    std::string tempfile = file.fullPath();
 
-    std::ifstream in_file(tempfile, std::ios::in);
+    std::ifstream in_file(tempfile.c_str(), std::ios::in);
```

Comment:
will it work if we pass ` file.fullPath().c_str()` directly here (without create tempfile string)?

Reply:
`tempfile` is used also in [`else` branch](https://github.com/cms-sw/cmssw/pull/46181/files/94fac8cb5212bf49aa7e75d4484fd4f72f0833bf#diff-a289dfa5c8d27f6de8b4833c4af79c340d16452c69953dd8ffe18a6ba4c985b4R86)

Reply:
frequency of hitting `else` is very low. So one can just use `file.fullPath()` there too ... right?

Reply:
> frequency of hitting `else` is very low. So one can just use `file.fullPath()` there too ... right?

I would think so.